src/Microsoft.Azure.WebJobs.Extensions.Kafka/KafkaListener.cs

```diff
@@ -0,0 +1,229 @@
+        private readonly string eventHubConnectionString;
+        private readonly string avroSchema;
+        private readonly int maxBatchSize;
+        private Consumer<TKey, TValue> consumer;
+        IKafkaMessagePublisher<TKey, TValue> publisher;
+        private bool disposed;
+
+        public KafkaListener(
+            ITriggeredFunctionExecutor executor,
+            bool singleDispatch,
+            ILogger logger,
+            string brokerList,
+            string topic,
+            string consumerGroup,
+            string eventHubConnectionString,
+            string avroSchema,
+            int maxBatchSize)
+        {
+            this.executor = executor;
+            this.singleDispatch = singleDispatch;
+            this.logger = logger;
+            this.brokerList = brokerList;
+            this.topic = topic;
+            this.consumerGroup = consumerGroup;
+            this.eventHubConnectionString = eventHubConnectionString;
+            this.avroSchema = avroSchema;
+            this.maxBatchSize = maxBatchSize;
+        }
+
+        public void Cancel()
+        {
+            this.SafeCloseConsumer();
+        }        
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            ConsumerConfig conf;
+            if (string.IsNullOrEmpty(this.eventHubConnectionString))
+            {
+                // Setup raw kafka configuration
+                conf = new ConsumerConfig
+                {
+                    BootstrapServers = this.brokerList,
+                    GroupId = consumerGroup,
+                    StatisticsIntervalMs = 5000,
+                    SessionTimeoutMs = 6000,
+                    AutoOffsetReset = AutoOffsetReset.Earliest,
+                    EnablePartitionEof = true,
+                    // { "debug", "security,broker,protocol" }       //Uncomment for librdkafka debugging information
+                };
+            }
+            else
+            {
+                var ehBrokerList = this.brokerList;
+                if (!ehBrokerList.Contains(".servicebus.windows.net"))
+                {
+                    ehBrokerList = $"{ this.brokerList}.servicebus.windows.net:9093";
+                }
+
+                var consumerGroupToUse = string.IsNullOrEmpty(this.consumerGroup) ? "$Default" : this.consumerGroup;
+
+                conf = new ConsumerConfig
+                {
+                    BootstrapServers = ehBrokerList,
+                    StatisticsIntervalMs = 5000,
+                    SessionTimeoutMs = 6000,
+                    AutoOffsetReset = AutoOffsetReset.Earliest,
+                    EnablePartitionEof = true,
+                    SecurityProtocol = SecurityProtocol.SaslSsl,
+                    SaslMechanism = SaslMechanism.Plain,
+                    SaslUsername = "$ConnectionString",
+                    SaslPassword = this.eventHubConnectionString,
+                    SslCaLocation = "./cacert.pem",
```

Comment:
Just curious, why do you need this?

Reply:
@fbeltrao ? are these needed, or just a carry over from copying the initial EventHub implementation?

Reply:
They are not needed, unless we want to support EventHubs w/ Kafka head during initial phase. If you don't mind, I can solve this in next PR.

Reply:
Great. Thanks. Let's remove them then. I imagine for EventHubs with Kafka head we would just use the same connection string / credentials that we used for Kafka?

Reply:
Linking issue #21 to this.